politeiawww/www.go

```diff
@@ -512,11 +511,18 @@ func _main() error {
 	p.router.Use(recoverMiddleware)
 
 	// Setup dcrdata websocket connection
-	ws, err := newWSDcrdata(p.dcrdataHostWS())
-	if err != nil {
-		return fmt.Errorf("new wsDcrdata: %v", err)
-	}
-	p.wsDcrdata = ws
+	go func() {
+		for {
+			ws, err := newWSDcrdata(p.dcrdataHostWS())
```

Comment:
This is likely going to cause panics in other parts of the code that rely on the `wsDcrdata` because the subscriptions map is not being initialized.  `wsdcrdata` has a `reconnect()` methods that was written to handle the case where dcrdata drops the connection.  Can we use that method instead?  politeiawww was written with the assumption that `wsDcrdata` might be trying to reconnect, so it shouldn't cause any unforeseen issues.  What would need to happen would be:

1. Update `newDcrdataWSClient` to return a error variable that can be keyed off of when the function fails due to a connection error.
2. Have `newWSDcrdata` returned the initialized `wsDcrdata` context along with the error.  Document why we chose to do this since it is a non-standard approach.
3. If `newWSDcrdata` fails due to the connection error, run the reconnect function in a go routine.